clippy_lints/src/utils/conf.rs

```diff
@@ -110,8 +110,10 @@ macro_rules! define_Conf {
 define_Conf! {
     /// Lint: BLACKLISTED_NAME. The list of blacklisted names to lint about
     (blacklisted_names, "blacklisted_names", ["foo", "bar", "baz", "quux"] => Vec<String>),
-    /// Lint: CYCLOMATIC_COMPLEXITY. The maximum cyclomatic complexity a function can have
-    (cyclomatic_complexity_threshold, "cyclomatic_complexity_threshold", 25 => u64),
+    /// Lint: COGNITIVE_COMPLEXITY. The maximum cognitive complexity a function can have
+    (cognitive_complexity_threshold, "cognitive_complexity_threshold", 25 => u64),
+    /// DEPRECATED LINT: CYCLOMATIC_COMPLEXITY. Use the Cognitive Complexity lint instead.
+    (cyclomatic_complexity_threshold, "cyclomatic_complexity_threshold", 11_111_011_111 => u64),
```

Comment:
Ah, now I understand your confusion with what I was suggesting. My suggestion with using the field directly made no sense, since the fields aren't an option, but defaulted. Rereading the macro expanded code, I think we can make the default value `None` and the type `Option<u64>`. Then we don't need magic numbers but you can check `is_some()` instead.

Sorry about that.

Reply:
Ah, I see now. Yes, I didn't know how to check it otherwise. I'm testing now a couple of changes that should (I think) let the same code work, but by defaulting to a `None` value instead of a magic number. The one thing I'm wondering is if using an `Option<u64>` will be the same for the toml parsing code as it was using a `u64` directly. We'll see :) 
Thank you for clarifying that ^^